StylableSwiftUI/Classes/StylistMatchingMode.swift

```diff
@@ -0,0 +1,75 @@
+//
+//  StylistMatchingMode.swift
+//  StylableSwiftUI
+//
+//  Created by Kerr Marin Miller on 18/06/2020.
+//
+
+import Foundation
+
+public protocol StylistMatchingMode {
+    func insert(styles: [Style], into: [Style]) -> [Style]
+    func firstMatch(styles: [Style], toIdentifier: StylistIdentifier) -> Style?
+}
+
+public struct AtomicDesign: StylistMatchingMode {
+
+    public init() { }
+
+    public func insert(styles: [Style], into existingStyles: [Style]) -> [Style] {
+        let new = existingStyles + styles
+        return new.sorted { $0.identifier < $1.identifier }
+    }
+
+    public func firstMatch(styles: [Style], toIdentifier: StylistIdentifier) -> Style? {
+        return styles.first { $0.identifier.matches(toIdentifier) }
+    }
+}
+
+public struct NaturalDesign: StylistMatchingMode {
+
+    public init() { }
+
+    public func insert(styles: [Style], into existingStyles: [Style]) -> [Style] {
+        return existingStyles + styles
+    }
+
+    public func firstMatch(styles: [Style], toIdentifier: StylistIdentifier) -> Style? {
+        return styles.filter { $0.identifier.matches(toIdentifier) }
+            .sorted { a, b in
+                a.identifier.score(against: toIdentifier) > b.identifier.score(against: toIdentifier)
+        }.first
+    }
+}
+
+extension StylistIdentifier {
```

Comment:
Should this be private - it's really internal to `NaturalDesign`?

On the same lines, I don't think we should make StylsIdentifier comparable anymore - the comparison only works if you're using AtomicDesign?